mat/dense_arithmetic_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright ©2013 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat_test
+
+import (
+	"fmt"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+func ExampleDense_Add() {
+	// Initialize two matrices, a and b
+	a := mat.NewDense(2, 2, []float64{1, 0, 1, 0})
+	b := mat.NewDense(2, 2, []float64{0, 1, 0, 1})
+
+	// Add a and b, placing the result into b
```

Comment:
Do you think it would be instructive here to also have a new matrix? Something like

```
// Add a and b, placing the result into c. Notice that the size is automatically adjusted when the receiver has zero size.
var c mat.Dense
c.Add(a, b)
```